api/envoy/extensions/filters/network/mysql_proxy/v3/mysql_proxy.proto

```diff
@@ -15,15 +18,68 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // MySQL Proxy :ref:`configuration overview <config_network_filters_mysql_proxy>`.
 // [#extension: envoy.filters.network.mysql_proxy]
 
+// Auth information of MySQL.
+message AuthInfo {
+  config.core.v3.DataSource auth_username = 1 [(udpa.annotations.sensitive) = true];
+
+  config.core.v3.DataSource auth_password = 2 [(udpa.annotations.sensitive) = true];
+}
+
+// [#next-free-field: 6]
 message MySQLProxy {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy";
 
+  message Route {
+    // String database name that must exact match with client connection db name
+    string database = 1 [(validate.rules).string = {min_len: 1}];
+
+    // Upstream cluster to forward the command to.
+    string cluster = 2 [(validate.rules).string = {min_len: 1}];
+  }
+
   // The human readable prefix to use when emitting :ref:`statistics
   // <config_network_filters_mysql_proxy_stats>`.
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // [#not-implemented-hide:] The optional path to use for writing MySQL access logs.
   // If the access log field is empty, access logs will not be written.
   string access_log = 2;
+
+  // [#not-implemented-hide:] enable manage MySQL protocol at proxy. Default is false.
+  // * When enable_manage_protocol is true, MySQL filter is a terminal filter, it's config should put at the last of filter chain, in the same time, MySQLProtocolOptions should be set at corresponding config of cluster for connecting to upstream database.
+  // * When enable_manage_protocol is false, MySQL filter is not a terminal filter, it's config can not put at the last of filter chain.
+  bool enable_manage_protocol = 3;
```

Comment:
Isn't this implied by whether `routes` has element or not? Existing behavior delegate route select to `tcp_proxy` so in that case route shouldn't have anything.

Reply:
Yes, when `enable_manage_protocol` is false, `route` and `auth_info` should not set value.